### PR TITLE
ARROW-6039: [GLib] Add garrow_array_filter()

### PR DESCRIPTION
**Pull request by shiro615**
This PR add support for boolean selection filter.

### CODE REVIEW

File: c_glib/arrow-glib/compute.cpp [hunk 1]
```diff
@@ -1377,6 +1377,40 @@ garrow_double_array_compare(GArrowDoubleArray *array,
...
+ *
+ * Returns: (nullable) (transfer full): The #GArrowArray filterd
+ *   with a boolean selection filter. Nulls in the filter will
+ *   result in nulls in the output.
+ *
+ * Since: 0.15.0
+ */
+GArrowArray *
+garrow_array_filter(GArrowArray *array,
+                    GArrowArray *filter,
```
Comment (kou) [inline c_glib/arrow-glib/compute.cpp]: Could you use `GArrowBooleanArray`?

File: c_glib/test/test-filter.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+class TestFilter < Test::Unit::TestCase
+  include Helper::Buildable
+
+  def test_filter
+    filter = build_boolean_array([false, true, true, false])
```
Comment (kou) [inline c_glib/test/test-filter.rb]: Could you also use a `nil` value in filter?

**Review by kou — CHANGES_REQUESTED** [review]
Thanks!
Could you check my comments?

File: c_glib/arrow-glib/compute.cpp (hunk 1 above)
Comment (shiro615) [reply c_glib/arrow-glib/compute.cpp]: I've fixed it.

File: c_glib/test/test-filter.rb (hunk 2 above)
Comment (shiro615) [reply c_glib/test/test-filter.rb]: I've included a nil value in filter.

**Review by kou — APPROVED** [review]
+1

**Pull request closed without merging**